tests/e2e/test_performance_fio_randread_profile.py

```diff
@@ -0,0 +1,105 @@
+"""
+Module to measure IO Latency with several weights
+"""
+import pytest
+import logging
+from ocs_ci.utility.spreadsheet.spreadsheet_api import GoogleSpreadSheetAPI
+from ocs_ci.framework.testlib import (
+    E2ETest, performance, google_api_required,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@google_api_required
+@performance
+class TestIORandReadProfile(E2ETest):
+    """
+    Test Random read IO performance using FIO
+    """
+    @pytest.fixture()
+    def base_setup(self, request, interface_iterate, pod_factory):
+        """
+        A setup phase for the test
+        Args:
+            interface_iterate: will iterate over RBD and CephFS interfaces
+                to parametrize the test
+            pod_factory: A fixture to create everything needed for a running
+                pod
+        """
+        self.interface = interface_iterate
+        self.pod_obj = pod_factory(self.interface)
+
+    @pytest.mark.parametrize(
+        argnames=[
+            "size", "io_direction", "jobs",
+            "runtime", "depth", "sheet_index"
+        ],
+        argvalues=[
+            pytest.param(
+                *['1G', 'ro', 2, 300, 64, 4]
+            ),
+            pytest.param(
+                *['1G', 'ro', 2, 300, 16, 4]
+            ),
+            pytest.param(
+                *['1G', 'ro', 2, 300, 4, 4]
+            ),
+            pytest.param(
+                *['10M', 'ro', 2, 300, 64, 3]
+            ),
+            pytest.param(
+                *['10M', 'ro', 2, 300, 16, 3]
+            ),
+            pytest.param(
+                *['10M', 'ro', 2, 300, 4, 3]
+            ),
+            pytest.param(
+                *['1M', 'ro', 2, 300, 64, 2]
+            ),
+            pytest.param(
+                *['1M', 'ro', 2, 300, 16, 2]
+            ),
+            pytest.param(
+                *['1M', 'ro', 2, 300, 4, 2]
+            ),
+            pytest.param(
+                *['10k', 'ro', 2, 300, 64, 1]
+            ),
+            pytest.param(
+                *['10k', 'ro', 2, 300, 16, 1]
+            ),
+            pytest.param(
+                *['10k', 'ro', 2, 300, 4, 1]
+            )
+        ]
+    )
+    @pytest.mark.usefixtures(base_setup.__name__)
+    def test_perf_read_latency(
+        self, size, io_direction, jobs, runtime, depth, sheet_index
+    ):
+        """
+        Test IO
```

Comment:
Function does not have enough description,i see it does a lot  of actions and it  will be better to have description and pass and fail criteria's

Reply:
I've added a description now. These tests as of today just generate graphs for the IO profiles. I'll introduce a check to fail the test once we have a benchmark to compare. I've added a todo for now.